src/SymbolCollector.Server/SymbolService.cs

```diff
@@ -155,11 +155,18 @@ public async Task<StoreResult> Store(Guid batchId, string fileName, Stream strea
             throw new InvalidOperationException("Couldn't get the path from tempDestination: " + tempDestination);
         }
 
-        Directory.CreateDirectory(path);
-        await using (var file = File.OpenWrite(tempDestination))
+        try
+        {
+            Directory.CreateDirectory(path);
+            await using (var file = File.OpenWrite(tempDestination))
+            {
+                await stream.CopyToAsync(file, token);
+                _logger.LogDebug("Temp file {bytes} copied {file}.", file.Length, Path.GetFileName(tempDestination));
+            }
+        }
+        catch (System.IO.IOException ioEx)
         {
-            await stream.CopyToAsync(file, token);
-            _logger.LogDebug("Temp file {bytes} copied {file}.", file.Length, Path.GetFileName(tempDestination));
+            _logger.LogInformation(ioEx, "Caught IOException during file write operation, likely due to concurrent access. Continuing normally.");
```

Comment:
we should be returning from here instead of continuing since the file wasn't successfuly written to disk